runtime/gc_base/RootScanner.cpp

```diff
@@ -536,7 +537,23 @@ MM_RootScanner::scanOneThread(MM_EnvironmentBase *env, J9VMThread* walkThread, v
 		doVMThreadSlot(slot, &vmThreadIterator);
 	}
 
-	GC_VMThreadStackSlotIterator::scanSlots((J9VMThread *)env->getOmrVMThread()->_language_vmthread, walkThread, localData, stackSlotIterator, isStackFrameClassWalkNeeded(), _trackVisibleStackFrameDepth);
+	J9VMThread *currentThread = (J9VMThread *)env->getOmrVMThread()->_language_vmthread;
+	GC_VMThreadStackSlotIterator::scanSlots(currentThread, walkThread, localData, stackSlotIterator, isStackFrameClassWalkNeeded(), _trackVisibleStackFrameDepth);
```

Comment:
You should probably put a comment above, something like this:
In a case, this thread is a carrier thread, and a virtual thread is mounted, we will scan virtual thread's stack.

Reply:
You can append this comment by another sentence.
... If virtual thread is not mounted, or this is just a regular thread, this will scan its own stack.